### ISSUE TITLE
bool-param-default should ignore type declaration when function has meet requirements

### ISSUE THREAD

**Issue by michaelw85** [SonarSource/SonarTS#857]
**I want to report a bug.**

**SonarTS version:** 1.9.0<br>
**Node.js version:** v10.15.1<br>
**TypeScript version:** 3.4.5<br>

**SonarQube version:** N/A<br>
**SonarLint version and used IDE:** No, vscode<br>
**TSLint version:** 5.16.0<br>

**Rule key:** 
`bool-param-default`
Provide a default value for 'XYZ' so that the logic of the function is more evident when this parameter is missing. Consider defining another function if providing default value is not possible.

**Reproducer** 
```ts
const configure: (config: FrameworkConfiguration, devMode?: boolean) => FrameworkConfiguration = (
    config: FrameworkConfiguration,
    devMode: boolean = false
): FrameworkConfiguration => // function logic
```
**Expected behavior**
Rule should not throw error for type declaration if function sets a default value.
